#!/usr/bin/env python
PACKAGE = "franka_interactive_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness_x", double_t, 0, "Cartesian translational stiffness x", 300, 0, 1500)
gen.add("translational_stiffness_y", double_t, 0, "Cartesian translational stiffness y", 300, 0, 1500)
gen.add("translational_stiffness_z", double_t, 0, "Cartesian translational stiffness z", 300, 0, 1500)

gen.add("rotational_stiffness_x", double_t, 0, "Cartesian rotational stiffness x", 20, 0, 50)
gen.add("rotational_stiffness_y", double_t, 0, "Cartesian rotational stiffness y", 20, 0, 50)
gen.add("rotational_stiffness_z", double_t, 0, "Cartesian rotational stiffness z", 20, 0, 50)

# gen.add("all_TRANSLATIONAL_stiffness", double_t, 0, "Cartesian TRANSLATIONAL stiffness", 200, 0, 400)
# gen.add("all_ROTATIONAL_stiffness", double_t, 0, "Cartesian ROTATIONAL stiffness", 10, 0, 30)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param_hybrid"))
